Update for Logs feature (#32767)
DataDog/documentation: content/en/developers/ide_plugins/idea/logs.md

---
title: Logs
type: documentation
further_reading:
- link: "/logs/explorer/"
  tag: "Documentation"
  text: "Learn more about Logs"
---

## Overview
The Logs integration detects log lines in your source code, displays live event counts directly in the source editor, and provides links to the Datadog [Log Explorer][16] to view the logs generated by each log line. 

{{< img src="/developers/ide_plugins/idea/logs/logs.png" alt="Logs in the source editor" style="width:100%;" >}}

Hover over a log element with your mouse pointer to see matching log patterns with runtime value ranges. Use this runtime context to quickly understand how your code behaves in production or other deployed environments.

> ## Log Queries

Reviewer: ```suggestion
## Log queries
```

Click the log icon to query logs using Datadog:

{{< img src="/developers/ide_plugins/idea/logs/log-query-button.png" alt="The log query button" style="width:80%;" >}}

This launches the [Log Explorer][16] and shows the latest log events.  Notice how the query has been prefilled with the environment, log status, logger name and text elements extracted from your source code:

{{< img src="/developers/ide_plugins/idea/logs/browser.png" alt="The log query button" style="width:100%;" >}}

> Use the full power of the Datadog platform to inspect individual logs and related traces, or modify the search query to narrow down to the logs that you are most interested in.

Reviewer: ```suggestion
Use the Datadog platform to inspect individual logs and related traces, or modify the search query to narrow down to the logs that you are most interested in.
```

> ## Supported Languages & Frameworks

Reviewer: ```suggestion
## Supported languages and frameworks
```

* Go : [Logrus][4], [Zap][5] and the [log package][6] in the Standard Library
* Python : [Python logging][7] and [Loguru][8]
* JavaScript & TypeScript : [Datadog Browser Logs][9] and [Winston][10]
* PHP : [Laravel][11], [Monolog][12], [Symfony][13] and [PSR-3][14] 

## Settings
Right-click a log element in the source editor to open the settings (Editor → Inlay Hints) for the current language, and activate or deactivate the feature.

### Java / Kotlin
In Java & Kotlin, by convention, logs are normally tagged with the logger name corresponding to the fully qualified class name of the class where the logger is created.  

In rare cases, customers post-process their logs data and modify the logger name tag to use only the simple class name. For these customers, there is a “Logger name” setting to inform the plugin to create Log Explorer queries using the simple class name.

{{< img src="/developers/ide_plugins/idea/logs/settings-java.png" alt="Log settings - Java" style="width:100%;" >}}

### Go, Python, JavaScript, TypeScript and PHP
For these languages, there are no custom settings, only the activate/deactivate checkbox.

## Advanced Settings

> For a normal setup, the advanced settings are not required. However, a few Datadog customers have log events with a non-standard [logger name attribute][15]. You can change the “Logger name tag” setting accordingly. If you set it to blank, the logger name will not be used in log queries at all.

Reviewer: ```suggestion
## Advanced settings

For a typical setup, the advanced settings are not required. However, if you have log events with a non-standard [logger name attribute][15], you can change the “Logger name tag” setting accordingly. If you set it to blank, the logger name will not be used in log queries at all.
```
